src/Inviqa/Worldpay/Api/PaymentAuthorizer.php

```diff
@@ -3,37 +3,63 @@
 namespace Inviqa\Worldpay\Api;
 
 use Inviqa\Worldpay\Api\Exception\ConnectionFailedException;
-use Inviqa\Worldpay\Api\Request\RequestFactory;
+use Inviqa\Worldpay\Api\Request\AuthorizeRequestFactory;
+use Inviqa\Worldpay\Api\Request\ThreeDSRequestFactory;
 use Inviqa\Worldpay\Api\Response\AuthorisedResponse;
 use Inviqa\Worldpay\Api\Response\ResponseFactory;
 
 class PaymentAuthorizer
 {
-    private $requestFactory;
+    private $authRequestFactory;
+    private $threeDSRequestFactory;
     private $xmlNodeConverter;
     private $client;
 
     public function __construct(
-        RequestFactory $requestFactory,
+        AuthorizeRequestFactory $authRequestFactory,
+        ThreeDSRequestFactory $threeDSRequestFactory,
         XmlNodeConverter $xmlNodeConverter,
         Client $client
     )
     {
-        $this->requestFactory = $requestFactory;
+        $this->authRequestFactory = $authRequestFactory;
+        $this->threeDSRequestFactory = $threeDSRequestFactory;
         $this->xmlNodeConverter = $xmlNodeConverter;
         $this->client = $client;
     }
 
     /**
      * @param array $paymentParameters
-     *
      * @return AuthorisedResponse
      * @throws ConnectionFailedException
      */
     public function authorizePayment(array $paymentParameters)
     {
-        $paymentService = $this->requestFactory->buildFromRequestParameters($paymentParameters);
+        $paymentService = $this->authRequestFactory->buildFromRequestParameters($paymentParameters);
 
+        return $this->makeRequest($paymentService);
+    }
+
+    /**
+     * @param array $paymentParameters
+     * @return AuthorisedResponse
+     * @throws ConnectionFailedException
+     * @throws Exception\InvalidRequestParameterException
+     */
+    public function authorize3DSecure(array $paymentParameters)
+    {
+        $paymentService = $this->threeDSRequestFactory->buildFromRequestParameters($paymentParameters);
+
+        return $this->makeRequest($paymentService);
```

Comment:
The client should also send a `Cookie` header with the machine cookie value.
The value of the cookie should be available in the `$paymentParameters` array, but the action of setting the header value should be performed by the client.